### ISSUE TITLE
Webapp import amount always 0

### ISSUE THREAD

**Issue by osr-revamper** [moneymanagerex/moneymanagerex#4277]
MMEX version:
 - [x] 1.5.x
 - [ ] 1.6.x

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

When trying to import new Webapp transactions in my MMEX Desktop, the "Amount" of all transactions is always 0. 

Tested with master, 1.0.3 and 1.0.2

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. Create new transactions in WebApp with Amount field completed with random imports.
2. Open Desktop MMEX
3. In the webapp import tool verify taht the amounts are always 0.

Expected result: The same amount in the webapp import tool than on the web server.

Actual result: All the amounts are always 0.

# Additional information

Tested with PHP 7.3, 7.4 and 8.0.
![mmex_desktop_sync](https://user-images.githubusercontent.com/96193896/161500040-30a4fd63-6430-4376-a0a5-9297fbde76c2.png)
![mmex_webapp_pending](https://user-images.githubusercontent.com/96193896/161500044-2fcea4f3-f08a-4508-886f-c723a68382db.png)

**Comment by whalley** [comment]
I can't reproduce this. Perhaps it is related to currency format/settings.

![CleanShot 2022-04-05 at 00 21 43](https://user-images.githubusercontent.com/487345/161648159-af3b58fa-f3a4-4f82-96bb-30fc7f299553.png)

![CleanShot 2022-04-05 at 00 22 04](https://user-images.githubusercontent.com/487345/161648222-5cb9c0f3-a224-4593-95bc-58006d28ced9.png)

**Comment by osr-revamper** [comment]
These are my settings:

- Desktop in Spanish locale.
- Accounts in EUR.
 
![mmex_1](https://user-images.githubusercontent.com/96193896/161705868-23fc7e2c-fb18-47df-8aa0-5a78fd0d1e5b.png)
![mmex_2](https://user-images.githubusercontent.com/96193896/161705878-0998a704-de18-49c4-982a-98ec3cc8a85e.png)
![mmex_3](https://user-images.githubusercontent.com/96193896/161705882-647295f1-b870-46e8-82cd-3b53822d9ede.png)
![mmex_4](https://user-images.githubusercontent.com/96193896/161705883-38279a4d-d8d3-4789-a7f1-ee8d6ababd30.png)
![mmex_5](https://user-images.githubusercontent.com/96193896/161705885-442fc7d0-a915-409a-a041-232047a4dc12.png)

**Comment by whalley** [comment]
Can you please enter a few new transactions into the WebApp the goto the WebApp install directory:

The should be a file in there called **MMEX_New_Transaction.db**

Now open the DB by typing:
`sqlite3 MMEX_New_Transaction.db `

You should be prompted with:

```
SQLite version 3.31.1 2020-01-27 19:55:54
Enter ".help" for usage hints.
sqlite> 
```

Just enter the command:
`select * from New_Transaction;`

and let me know the output.

**Comment by osr-revamper** [comment]
Here you are:

SQLite version 3.31.1 2020-01-27 19:55:54
Enter ".help" for usage hints.
sqlite> select * from New_Transaction;
2|2022-04-04|ING |None|F|Withdrawal|None|None|None|12.36|Testing
3|2022-04-06|ING |None||Withdrawal|None|None|None|23.45|
4|2022-04-06|Caja |None||Withdrawal|None|None|None|56.89|pruebas
5|2022-04-06|Caja |None|D|Transfer|None|None|None|23|eee
6|2022-04-06|BBVA |None|R|Deposit|None|None|None|12.67|

Thanks whalley!

**Comment by whalley** [comment]
@osr-revamper Thanks for sharing your server detail... So I can see why the issue is arising...

My WebApp returns data like....
```
"0": {
    "ID": "2",
    "Date": "2022-04-05",
    "Account": "Cash (UK)",
    "ToAccount": "None",
    "Status": "",
    "Type": "Withdrawal",
    "Payee": "1Passwd",
    "Category": "Adjustments",
    "SubCategory": "",
    "Amount": "12.34",
    "Notes": "",
    "Attachments": ""
}
```

Yours returns:
```
"0": {
    "ID": 2,
    "Date": "2022-04-04",
    "Account": "ING 7505",
    "ToAccount": "None",
    "Status": "F",
    "Type": "Withdrawal",
    "Payee": "None",
    "Category": "None",
    "SubCategory": "None",
    "Amount": 12.359999999999999,
    "Notes": "Testing",
    "Attachments": ""
}
```

Note the difference in the presentation of the **Amount** field. The MMEX application is always expecting a STRING for each field but your WebApp installation is returning a numeric value.

Can you confirm you have the latest WebApp Version 1.1.0 / Master code installed?

At least we have narrowed this down.....

**Comment by osr-revamper** [comment]
My PHP version is 8.1.

**Comment by osr-revamper** [comment]
You are right whalley. From PHP 7.1 till 8.0 is working now!! But not in PHP 8.1.

Many thanks for your help.

Best regards.

**Comment by whalley** [comment]
@osr-revamper Can you please try the following for me.

Edit the functions.php file, to add the following line:

`            $db->setAttribute(PDO::ATTR_STRINGIFY_FETCHES, true);`

after line 461, e.g.

```
    // Select all transaction order by date
    public static function transaction_select_all_order_by_date (String $s_direction = 'DESC')
        {
            $const_dbpath = costant::database_path();
            $db = new PDO("sqlite:${const_dbpath}");
            $db->setAttribute(PDO::ATTR_STRINGIFY_FETCHES, true);
            
            $results = $db -> query("SELECT * FROM New_Transaction ORDER BY Date $s_direction, id $s_direction;");
            $resultarray = array();
            if($results !== false)
                {$resultarray = $results->fetchall(PDO::FETCH_ASSOC);}
            $db = null;
            return $resultarray;
        }
```
Then test with PHP 8.1 again

I think (hope)this should cure the problem.

**Comment by whalley** [comment]
Great, I'll move the issue over to the WebApp repo and make the change.
